compiler/rustc_middle/src/dep_graph/dep_node.rs

```diff
@@ -175,8 +203,16 @@ pub mod dep_kind {
                 const is_anon: bool = contains_anon_attr!($($attrs)*);
                 const is_eval_always: bool = contains_eval_always_attr!($($attrs)*);
 
+                #[inline(always)]
+                fn can_reconstruct_query_key() -> bool {
+                    !is_anon &&
```

Comment:
Why are we checking is_anon here? it seems like that's done in the caller of this already?

That said I see we're calling this directly in future commits, but seems good to still unify on where that check should happen.